tablecodec/tablecodec_test.go

```diff
@@ -64,31 +64,37 @@ func (s *testTableCodecSuite) TestRowCodec(c *C) {
 	c1 := &column{id: 1, tp: types.NewFieldType(mysql.TypeLonglong)}
 	c2 := &column{id: 2, tp: types.NewFieldType(mysql.TypeVarchar)}
 	c3 := &column{id: 3, tp: types.NewFieldType(mysql.TypeNewDecimal)}
-	cols := []*column{c1, c2, c3}
+	c4 := &column{id: 4, tp: &types.FieldType{Tp: mysql.TypeEnum, Elems: []string{"a"}}}
+	c5 := &column{id: 5, tp: &types.FieldType{Tp: mysql.TypeSet, Elems: []string{"a"}}}
+	c6 := &column{id: 6, tp: &types.FieldType{Tp: mysql.TypeBit, Flen: 8}}
+	cols := []*column{c1, c2, c3, c4, c5, c6}
 
-	row := make([]types.Datum, 3)
+	row := make([]types.Datum, 6)
 	row[0] = types.NewIntDatum(100)
 	row[1] = types.NewBytesDatum([]byte("abc"))
 	row[2] = types.NewDecimalDatum(types.NewDecFromInt(1))
+	row[3] = types.NewMysqlEnumDatum(types.Enum{Name: "a", Value: 0})
+	row[4] = types.NewDatum(types.Set{Name: "a", Value: 0})
+	row[5] = types.NewDatum(types.BinaryLiteral{100})
 	// Encode
-	colIDs := make([]int64, 0, 3)
+	colIDs := make([]int64, 0, len(row))
 	for _, col := range cols {
 		colIDs = append(colIDs, col.id)
 	}
-	sc := &stmtctx.StatementContext{TimeZone: time.UTC}
+	sc := &stmtctx.StatementContext{TimeZone: time.Local}
```

Comment:
why was this changed?

Reply:
Cover here https://github.com/pingcap/tidb/blob/master/tablecodec/tablecodec.go#L255